Tabloid/Repositories/PostRepository.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections.Generic;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Configuration;
+using Tabloid.Models;
+using Tabloid.Utils;
+
+namespace Tabloid.Repositories
+{
+    public class PostRepository : BaseRepository, IPostRepository
+    {
+        public PostRepository(IConfiguration configuration) : base(configuration) { }
+
+        public List<Post> GetAllPublishedPosts()
+        {
+            using (var conn = Connection)
+            {
+                conn.Open();
+                using (var cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"
+                        SELECT p.Id, p.Title, p.Content,
+                               p.ImageLocation AS HeaderImage,
+                               p.CreateDateTime, p.PublishDateTime,         p.IsApproved,
+                               p.CategoryId, p.UserProfileId,
+
+                               u.FirstName, u.LastName, u.DisplayName,
+                               u.Email, u.CreateDateTime, u.ImageLocation AS AvatarImage,
+                               u.UserTypeId,
+                           
+                               ut.[Name] AS UserTypeName
+                          FROM Post p
+                               LEFT JOIN UserProfile u ON p.UserProfileId = u.Id
+                               LEFT JOIN UserType ut ON u.UserTypeId = ut.Id
+                         WHERE IsApproved = 1 AND PublishDateTime < SYSDATETIME()
+                      ORDER BY PublishDateTime DESC";
+                    var reader = cmd.ExecuteReader();
+
+                    var posts = new List<Post>();
+
+                    while (reader.Read())
+                    {
+                        posts.Add(NewPostFromReader(reader));
+                    }
+
+                    reader.Close();
+
+                    return posts;
+                }
+            }
+        }
+
+        private Post NewPostFromReader(SqlDataReader reader)
```

Comment:
This function is one we haven't seen as a group, so it would be nice to leave a descriptive comment for your teammates to know exactly what it's purpose is and how to use it.

Reply:
Added.